api-client/spec/api/extensions/get_into_teaching_api_client/api_client_spec.rb

```diff
@@ -171,4 +172,90 @@ def perform_post_request
 
     expect { perform_get_request }.to raise_error(Faraday::ConnectionFailed)
   end
+
+  describe "Circuit breaker" do
+    let(:threshold) { 3 }
+    let(:timeout) { 5.minutes }
+
+    context "when enabled" do
+
+      before(:each) do
+        Stoplight::Light.default_data_store = Stoplight::DataStore::Memory.new
+
+        GetIntoTeachingApiClient.configure do |config|
+          config.circuit_breaker = { enabled: true, threshold: threshold, timeout: timeout }
+        end
+      end
+
+      context "when the API returns a single error that can cause a broken circuit" do
```

Comment:
May be worth making this "when the API returns (threshold - 1) errors that can cause a broken circuit" for completeness?